viewer/packages/360-images/src/Image360Icon.ts

```diff
@@ -2,173 +2,118 @@
+  set visible(visible: boolean) {
+    const alpha = visible ? 255 : 0;
+    this._alphaAttributeAccessor.set([alpha]);
+  }
 
-    sceneHandler.addCustomObject(this);
+  get visible(): boolean {
+    const alpha = this._alphaAttributeAccessor.at(0)!;
+    this._renderHook.visible = alpha > 0;
+    return alpha > 0;
   }
 
   set hoverSpriteVisible(visible: boolean) {
     this._hoverSprite.visible = visible;
   }
 
-  public raycast(raycaster: THREE.Raycaster, intersects: THREE.Intersection<THREE.Object3D<THREE.Event>>[]): void {
-    if (this.visible === false) {
-      return;
-    }
-    const intersections: THREE.Intersection[] = [];
-    this._outerSprite.raycast(raycaster, intersections);
-    if (intersections.length > 0) {
-      const obj = intersections[0];
-      obj.object = this;
-      intersects.push(obj);
-    }
+  public intersect(ray: Ray): Vector3 | null {
+    const sphere = new Sphere(this._position, 0.5 * this._adaptiveScale);
+    return ray.intersectSphere(sphere, new Vector3());
   }
 
   public dispose(): void {
-    this._sceneHandler.removeCustomObject(this);
-
-    this._hoverSprite.material.map?.dispose();
+    this._sceneHandler.removeCustomObject(this._hoverSprite);
     this._hoverSprite.material.dispose();
     this._hoverSprite.geometry.dispose();
 
-    this._outerSprite.material.map?.dispose();
-    this._outerSprite.material.dispose();
-    this._outerSprite.geometry.dispose();
+    this._alphaAttributeAccessor.set([0]);
+    this._sceneHandler.removeCustomObject(this._renderHook);
+    this._renderHook.onBeforeRender = () => {};
   }
 
-  private setupAdaptiveScaling(modelToWorldTransform: THREE.Matrix4): void {
-    const base = new THREE.Vector4();
-    const offset = new THREE.Vector4();
-    const transform = new THREE.Matrix4();
-
-    this._outerSprite.onBeforeRender = (renderer: THREE.WebGLRenderer, _1: THREE.Scene, camera: THREE.Camera) => {
-      const adaptiveScale = computeAdaptiveScaling(renderer, camera, this.MAX_PIXEL_SIZE, this.MIN_PIXEL_SIZE);
-      this._outerSprite.scale.set(adaptiveScale, adaptiveScale, 1.0);
-      this._outerSprite.updateMatrixWorld();
-    };
-
-    this._hoverSprite.onBeforeRender = (renderer: THREE.WebGLRenderer, _1: THREE.Scene, camera: THREE.Camera) => {
-      const adaptiveScale = computeAdaptiveScaling(renderer, camera, this.MAX_PIXEL_SIZE, this.MIN_PIXEL_SIZE);
-      this._hoverSprite.scale.set(adaptiveScale, adaptiveScale, 1.0);
+  private setupAdaptiveScaling(position: THREE.Vector3, sceneHandler: SceneHandler, renderHook: Mesh): void {
+    const ndcPosition = new THREE.Vector4();
+    const renderSize = new THREE.Vector2();
+    renderHook.onBeforeRender = (renderer: THREE.WebGLRenderer, _1: THREE.Scene, camera: THREE.Camera) => {
+      this._adaptiveScale = computeAdaptiveScaling(renderer, camera, this._maxPixelSize, this._minPixelSize);
+      this._hoverSprite.scale.set(this._adaptiveScale, this._adaptiveScale, 1.0);
```

Comment:
So we have one threejs object per icon, which have an `onBeforeRender`. Why can't this be a single callback on a single object instead?